scheduler-server/scheduling-dsl-compiler/src/libs/scheduler-ast.ts

```diff
@@ -0,0 +1,70 @@
+export type U<BitLength extends 8 | 16 | 32 | 64> = number;
+export type U8 = U<8>;
+export type U16 = U<16>;
+export type U32 = U<32>;
+export type U64 = U<64>;
+export type I<BitLength extends 8 | 16 | 32 | 64> = number;
+export type I8 = I<8>;
+export type I16 = I<16>;
+export type I32 = I<32>;
+export type I64 = string;
+export type VarString<PrefixBitLength extends number, MaxBitLength extends number> = string;
+export type F<BitLength extends 32 | 64> = number;
+export type F32 = F<32>;
+export type F64 = F<64>;
+export type Duration = F64;
+export type Integer = I32;
+export type Double = F64;
+export type Time = string;
```

Comment:
Done. Upon further reflection, we probably don't need all of these types - my impression is that numeric types at the scheduler level will be simpler (basically just `Integer` and `Double`), unlike the command dictionaries which care a lot about how many bytes the numbers take up

Reply:
Agreed. If we don't need adherence on the scheduler side of things, thats fine. We should check number ranges coming back for integers to ensure they are between `Number.MIN_SAFE_INTEGER` and `Number.MAX_SAFE_INTEGER` to avoid any issues of unexpected truncation/rounding